docker-entrypoint.sh

```diff
@@ -20,6 +20,26 @@ else
   export NGINX_DEFAULT_SERVER_PORT="${NGINX_DEFAULT_SERVER_PORT:-80}"
 fi
 
+# Activate HSTS header (default: off)
+# https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+# The suggested value for the max-age is 63072000 (2 years).
+export NGINX_HSTS_MAX_AGE="${NGINX_HSTS_MAX_AGE:-0}"
+export NGINX_HSTS_INCLUDE_SUBDOMAINS="${NGINX_HSTS_INCLUDE_SUBDOMAINS:-1}"
+export NGINX_HSTS_PRELOAD="${NGINX_HSTS_PRELOAD:-1}"
+if [ "${NGINX_HSTS_MAX_AGE}" -gt 0 ]; then
+  export NGINX_HSTS_HEADER="max-age=${NGINX_HSTS_MAX_AGE}"
+  if [ "${NGINX_HSTS_INCLUDE_SUBDOMAINS}" -eq 1 ]; then
+    export NGINX_HSTS_HEADER="${NGINX_HSTS_HEADER}; includeSubDomains"
+  fi
+  if [ "${NGINX_HSTS_PRELOAD}" -eq 1 ]; then
+    export NGINX_HSTS_HEADER="${NGINX_HSTS_HEADER}; preload"
+  fi
+  sed -e '/#hstsheader/r /templates/hsts.conf' -i /templates/default.conf;
+  sed -e '/#hstsheader/r /templates/hsts.conf' -i /templates/subfolder.conf;
+  sed -e '/#hstsheader/r /templates/hsts.conf' -i /templates/catch-all-server.conf;
+  sed -e '/#hstsheader/r /templates/hsts.conf' -i /templates/from-to-www.conf.tpl;
+fi
```

Comment:
why having the fragment isn't enough?

Reply:
I guess i got it, the problem is that, fragments are actually loaded only by:

1. default.conf
2. subfolder.conf

And not by other server definitions like:

1. catch-all-server.conf
2. from-to-www.conf

That's why you are expanding this approach by using a sed substitution.

I am wondering how we can scale this approach and if we should instead just rely on fragments and assume that all Nginx servers defined by us inherit the fragments as well, even tho some fragments (like osb.conf) are not really needed nor useful in *not-apps* nginx server, like catch-all-server.

Reply:
Anyway the implementation is quite straightforward now, i ask you to:

1. Update the CHANGELOG file - use this space to explain better what is changing here
2. Update the PR title to make it clearer ("Add hosts header on redirects" is not very related)